assessment/collectors/app_migration_analysis.go

```diff
@@ -0,0 +1,499 @@
+						c. If it's a utility class, determine if any of its functionalities are affected by the new persistence layer.
+		4. Consider potential impacts on business logic or data flow due to changes in the underlying architecture.
+		5. Ensure the returned JSON is valid and parsable.
+		6. Capture larger code snippets for modification and provide cumulative descriptions instead of line-by-line changes.
+		7. Classify complexity as SIMPLE, MODERATE, or COMPLEX based on implementation difficulty, required expertise, and clarity of requirements.
+
+
+		**INPUT:**
+		File Path: %s
+		File Content:
+		%s
+		Method Changes:
+		%s`, filepath, content, *methodChanges)
+}
+
+func readFile(filepath string) (string, error) {
+	file, err := os.Open(filepath)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	var content string
+	for scanner.Scan() {
+		content += scanner.Text() + "\n"
+	}
+	if err := scanner.Err(); err != nil {
+		return "", err
+	}
+	return content, nil
+}
+
+func main() {
+	logger.Log = zap.NewNop()
+	ctx := context.Background()
+	projectID := "span-cloud-testing"
+	location := "us-central1"
+	apiKey := "<API_KEY>"
+	filePath := "/Users/pratick/IdeaProjects/GoLang-Gin-CRUD-App-using-MySQL/app/repositories/user_repository.go"
+
+	mysqlSchemaPath := "/Users/pratick/IdeaProjects/GoLang-Gin-CRUD-App-using-MySQL/my_queries.sql"
+	spannerSchemaPath := "/Users/pratick/IdeaProjects/GoLang-Gin-CRUD-App-using-MySQL/spanner.sql"
+	mysqlSchema, err := readFile(mysqlSchemaPath)
+	if err != nil {
+		fmt.Println("Error reading MySQL schema file:", err)
+		return
+	}
+
+	spannerSchema, err := readFile(spannerSchemaPath)
+	if err != nil {
+		fmt.Println("Error reading Spanner schema file:", err)
+		return
+	}
+	summarizer, err := NewMigrationSummarizer(ctx, &apiKey, projectID, location, mysqlSchema, spannerSchema)
+	if err != nil {
+		log.Fatalf("Error initializing summarizer: %v", err)
+	}
+
+	_, result := summarizer.AnalyzeFile(ctx, filePath, "")
+
+	fmt.Println(result)
+
+	var firstResult map[string]interface{}
+	err = json.Unmarshal([]byte(result), &firstResult)
+	if err != nil {
+		fmt.Println("Error parsing result:", err)
+		return
+	}
+
+	publicMethod, erra := json.MarshalIndent(firstResult["public_method_changes"], "", "  ")
+	if erra != nil {
+		fmt.Println("Error converting to struct:", erra)
+	}
+	fmt.Println(publicMethod)
+
+	secondFilePath := "/Users/pratick/IdeaProjects/GoLang-Gin-CRUD-App-using-MySQL/app/controllers/user_controller.go"
+	_, result = summarizer.AnalyzeFile(ctx, secondFilePath, string(publicMethod))
+
+	fmt.Println(result)
+}
+
+func getPromptForDAOClass(content, filepath string, methodChanges, oldSchema, newSchema *string) string {
+	return fmt.Sprintf(`
+        You are a Cloud Spanner expert tasked with migrating a DAO class from MySQL go-sql-mysql to Spanner go-sql-spanner.
```

Comment:
nit: it might make sense to move the prompts into prompt files. Please add a todo here